CONTRIBUTING.rst

```diff
@@ -1,18 +1,20 @@
 How to release
 --------------
 
-- Update CHANGELOG.rst
-- bump version (YYYY.MM) in pyproject.toml
-- commit
-- push to check one last time if the tests pass github side.
-- tag it (YYYY.MM).
-- build (``python -m build``)
-- Test it (in :file:`cpython/Doc` run
-  ``./venv/bin/pip install ../../python-docs-theme/dist/python-docs-theme-2021.8.tar.gz``
-  then build the doc using ``make html``).
-- upload it: ``twine upload dist/*``.
-- push the tag (``git push --tags``)
-
+- Update ``CHANGELOG.rst``
+- Bump version (YYYY.MM) in ``pyproject.toml``
+- Commit
+- Push to check tests pass on
+  `GitHub Actions <https://github.com/python/python-docs-theme/actions>`__
+- Go to https://github.com/python/python-docs-theme/releases
+- Click "Draft a new release"
+- Click "Choose a tag"
+- Type the next YYYY.MM version and
+  select "**Create new tag: YYYY.MM** on publish"
```

Comment:
Is it necessary to also specify the release title here or is it filled automatically after creating the tag?
![image](https://github.com/python/python-docs-theme/assets/25624924/bca575ad-0d12-47c4-9423-890c94fd429a)

Reply:
I left it blank and it automatically used the tag name.

I'll update the instructions to say to leave it blank.

I've just noticed I accidentally tagged it 2023.09 (leading zero) but the version in the package is 2023.9 (no leading zero). Earlier tags are without leading zero.

I'll also update the instructions to say no leading zero, and will also retag the repo.

(A second bit of automation would be to use setuptools_scm or hatch-vcs to single-source the version number, and skip some extra steps here. This would also enable Test PyPI uploads.)

Reply:
* [x] Checklist updated
* [x] 2023.09 tag deleted
* [x] 2023.9 tag added
* [x] Release updated to use 2023.9